Crossref support

Engerrs opened the pull request:
Hi guys

Added support for Crossref.

Out of the box it shouldn't break any original behavior, but also add an ability to use Crossref instead of DataCite.

Please review this and let me know if any adjustments are needed to be done.

Also updated the DB table initialization to use more modern CKAN way using migrations.

Engerrs committed:
Adding Crossref support

Engerrs committed:
add missing dependency

alycejenni reviewed (changes requested):
Hi @Engerrs, thank you for your PR! This is something we've considered adding in the past, but since we don't use crossref ourselves it hasn't been a priority.

I'd like to request a few changes:

- please make sure you read the [contributing guidelines](https://github.com/NaturalHistoryMuseum/ckanext-doi/blob/main/CONTRIBUTING.md), especially for code style and commit messages (for future commits)
- requirements need to go in `pyproject.toml`, not `requirements.txt`
- this is quite a significant new feature, so we need some test coverage for it

alycejenni committed:
ci: update workflow files
standardise format, change name of tests file

alycejenni committed:
ci: add pull request validation workflow
new workflow to check commit format and code style against pre-commit config

alycejenni committed:
ci: update tool config
update pre-commit repo versions and switch black to ruff

alycejenni committed:
chore: update tool details in contributing guide

alycejenni committed:
chore: add pull request template

alycejenni committed:
build: remove version from docker compose file
version specifier is deprecated

alycejenni committed:
build: update ckantools version

accidentally downgraded it automatically

alycejenni committed:
ci: only apply auto-fixes in pre-commit
F401 returns linting errors as well as auto-fixes, so this disables the errors and just applies the fixes

alycejenni committed:
ci: add docformatter args and dependency
docformatter currently can't read from pyproject.toml without tomli

alycejenni committed:
style: automatic reformat
auto reformat with ruff/docformatter/prettier after config changes

alycejenni committed:
docs: standardise returns field

alycejenni committed:
merge: #112 from ginger/standardise

Engerrs commented:
Hi @alycejenni , ok, I will adjust the code and write test for this.

alycejenni committed:
ci: add merge to valid commit types

alycejenni committed:
merge: #113 from ginger/merge-commit-type

alycejenni committed:
ci: fix python setup action version

alycejenni committed:
merge: #114 from ginger/python-setup-version

alycejenni committed:
merge: PR #115 from dev

Weekly release 2024-11-04

github-actions[bot] committed:
bump: version 3.1.15 → 3.1.16

alycejenni committed:
docs: fix tests badge
tests workflow file was renamed

alycejenni committed:
docs: use variable logo based on colour scheme

alycejenni committed:
merge: #116 from ginger/readme

alycejenni committed:
merge: #117 from patch

github-actions[bot] committed:
bump: version 3.1.16 → 3.1.17

alycejenni committed:
refactor(db)!: switch to alembic migration scripts

Gets rid of the old initdb command and introduces an initialisation migration script. Enables compatibility with CKAN 2.11.

BREAKING CHANGE: switch to alembic migration scripts

Closes: #118

alycejenni committed:
merge: #119 from ginger/2.11

alycejenni committed:
merge: PR #120 from dev

Weekly release 2025-03-17

github-actions[bot] committed:
bump: version 3.1.17 → 4.0.0

alycejenni committed:
ci: update pre-commit repo versions

alycejenni committed:
build: update ckantools

alycejenni committed:
build: remove pylint, add ruff lint ignore rules

alycejenni committed:
build: add ruff lint select rules

alycejenni committed:
ci: set ruff target py version, add more ignores
- avoid using fixes that don't work for python 3.8 (our current version)
- ignore recommended ruff formatter conflicts
- ignore more docstring rules

alycejenni committed:
merge: #121 from ginger/standardisation

alycejenni committed:
merge: PR #122 from dev

Weekly release 2025-06-09

github-actions[bot] committed:
bump: version 4.0.0 → 4.0.1

alycejenni committed:
fix: remove auto-added pyproject fields

alycejenni committed:
merge: #123 from patch

github-actions[bot] committed:
bump: version 4.0.1 → 4.0.2

alycejenni committed:
build: include db migration files

alycejenni committed:
merge: #124 from ginger/include-migration

alycejenni committed:
merge: PR #125 from dev

Weekly release 2025-08-11

github-actions[bot] committed:
bump: version 4.0.2 → 4.0.3

Engerrs committed:
fix(conflict): make sure crossref branch is up to date with main

Engerrs committed:
fix(tests): fix api tests

Engerrs committed:
fix: move dependency to pyproject

Engerrs committed:
fix: remove duplicated migration file
